### PR TITLE
Feature/cross env sessions

### PR DESCRIPTION
**Pull request by cdaringe**
#### asana

https://app.asana.com/0/10663154302695/68059429222401
# problem statement
- need to set auth cookie to support x-env login
- no tests
# solution
- mandate authCookieName on construction
- add tests.  the test set now is admittedly a little weak.  need to beef it up!  however, good basic assurances are met
- node_boilderplate align _most_ of the things.  i know there's mixed feelings on some of the boilerplate conventions.  let's make sure to get the boilerplate rev'd if any updates are needed.
# test
1. pull
2. npm i
3. `grunt test`
#### ps

saturday night. ya, i know--don't judge me! :) biked all afternoon and worked a pretty minimal day on friday.  here's to a hint of catch-up!

screen shot 2015-11-21 at 9 46 02 pm

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -27,3 +27,4 @@ build/Release
 node_modules
 
 .tmp
+test.bundle.js
```
Comment (cdaringe) [inline .gitignore:30]: this file get's generated to-be-injected into each browser by the testem runner

File: grunt/postcss.js [hunk 2]
```diff
@@ -2,7 +2,9 @@
 module.exports = function(grunt) {
     return {
         options: {
-            map: true, // Inline sourcemap
+            map: true,
+
+            // Inline sourcemap
```
Comment (cdaringe) [inline grunt/postcss.js:7]: .    >:( classic jscs

File: launch-tests.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        errorDie({ message: 'mock-api-server exited', code: code });
+    }
+};
+var testemExit = function(code) {
+    console.log('test run complete, closing server (exit code ' + code + ')');
+    server.kill();
+    return closeChild(code);
+};
+var spawnTestem = function() {
+    testem = cp.spawn('testem', ['ci', '-p', '7575', '-P', '5'], { stdio: 'inherit' });
```
Comment (cdaringe) [inline launch-tests.js]: `-P 5` => 5 concurrent browsers, `stdio: 'inherit'` => make the child process' stdxxx pipe directly into the parent's

File: scripts/lib/polyfills.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+module.exports = (function() {
+    if (!Function.prototype.bind) {
+        Function.prototype.bind = require('function-bind');
+    }
+})();
```
Comment (cdaringe) [inline scripts/lib/polyfills.js:5]: phantomJS polyfill*

**Comment by swashcap** [comment]
Running `grunt test` doesn’t run the tests on my machine. I get:

```
Running "exec:test" (exec) task
Spawned mock-api-server, pid: 12574
mock-api-server listening on 12354
server up! booting testem...
Spawned testem, pid: undefined
Failed to start child process { [Error: spawn testem ENOENT]
  code: 'ENOENT',
  errno: 'ENOENT',
  syscall: 'spawn testem',
  path: 'testem',
  spawnargs: [ 'ci', '-p', '7575', '-P', '5' ] }
```

Is this because testem needs to be globally installed?

File: examples/index.html [hunk 5]
```diff
@@ -55,7 +55,7 @@
             .example-form > * {
                 max-width: 300px;
             }
-            .example-code textarea {
+            .example-code pre {
```
Comment (swashcap) [inline examples/index.html:58]: Why the change to `pre`s? `textarea`s allow user focus, so one can easily select the entire element’s contents.

File: grunt/postcss.js (hunk 2 above)
Comment (swashcap) [inline grunt/postcss.js:7]: We should switch to ESLint at some point. Not necessary right now, of course.

File: scripts/lib/polyfills.js (hunk 4 above)
Comment (swashcap) [inline scripts/lib/polyfills.js:5]: `Function.bind` is ES5. Sheesh!

File: test/index.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+test('constructor, basic', function(t) {
+    var el = document.getElementById('regular_form');
+    var myWidget = new CoinsLogonWidget({
+        el: el,
+        baseUrl: 'coins-api-root',
+        authCookieName: 'coins-cookie-nom-noms',
+    });
+    t.ok(myWidget, 'constructor returns a widget');
+    t.equal(myWidget.element, el, 'constructor sets el');
+    t.ok(jQuery('#regular_form:visible').length, 'form visible on DOM');
```
Comment (swashcap) [inline test/index.js]: This checks that `<div id="regular_form">...</div>` is visible. Change to `#regular_form .coins-logon-widget:visible` or `#regular_form > *:visible`?

**Comment by swashcap** [comment]
Overall, great work! Just a couple notes.

File: test/index.js (hunk 6 above)
Comment (cdaringe) [inline test/index.js]: whoops! thanks!  kind of a silly test, but hey :)

**Comment by swashcap** [comment]
Tests work. _Woo!_
